app/src/main/java/org/p2p/wallet/home/ui/main/HomePresenter.kt

```diff
@@ -217,17 +221,54 @@ class HomePresenter(
             if (canBeClaimed) {
                 view?.showTokenClaim(token)
             } else {
-                val latestBundleId = token.latestActiveBundleId ?: return@launch
-                val latestBundleDetails = ethereumInteractor.getProgressDetails(latestBundleId)
-                transactionManager.emitTransactionState(latestBundleId, TransactionState.ClaimProgress(latestBundleId))
-                view?.showProgressDialog(
-                    bundleId = latestBundleId,
-                    progressDetails = latestBundleDetails ?: return@launch
-                )
+                token.latestActiveBundleId?.let { latestBundleId ->
+                    val latestBundleDetails = ethereumInteractor.getProgressDetails(latestBundleId)
+                    transactionManager.emitTransactionState(
+                        latestBundleId,
+                        TransactionState.ClaimProgress(latestBundleId)
+                    )
+                    if (latestBundleDetails != null) {
+                        showClaimProgressDetails(latestBundleId, latestBundleDetails)
+                    } else {
+                        tryGetProgressByToken(token)
+                    }
+                } ?: tryGetProgressByToken(token)
             }
         }
     }
 
+    private suspend fun tryGetProgressByToken(token: Token.Eth) {
+        ethereumInteractor.getBundleByToken(token)?.let { bridgeBundle ->
+            getProgressDetails(token, bridgeBundle)
+        } ?: Timber.i("No bundle found associated to token: $token to show claim details!")
```

Comment:
should we log as .e?

Reply:
maybe
i do not know how much this is important
but on lets do it with `.e`

Reply:
TImber.e(IllegalStateException("No bundle found associated to token: $token to show claim details!")) will be ok for now